Calibration/EcalCalibAlgos/python/ecalPedestalPCLworker_cfi.py

```diff
@@ -5,7 +5,7 @@
                                 BarrelDigis=cms.InputTag('ecalDigis','ebDigis'),
                                 EndcapDigis=cms.InputTag('ecalDigis','eeDigis'),
                                 tcdsRecord  =cms.InputTag('tcdsDigis','tcdsRecord'),
-                                requireStableBeam = cms.bool(True),
+                                requireStableBeam = cms.bool(False),
```

Comment:
wildly speculating: can this be included in the cosmics reconstruction scenario, which by construction doesn't have stable beams, without having to pollute conditions with unstable collisions?
https://github.com/cms-sw/cmssw/blob/master/Configuration/DataProcessing/python/Impl/cosmics.py
Alternatively one can adapt the C++ source code here:
https://github.com/cms-sw/cmssw/blob/c861ecffe8b371654f4f0e917036d83a0801f07e/Calibration/EcalCalibAlgos/src/ECALpedestalPCLworker.cc#L48-L54
to accept also `BSTRecord::BeamMode::NOBEAM ` (I suppose that's what the TDSC record has for cosmics).

Not sure though one wants to run the ecal pedestal on *all* cosmics runs.

Reply:
From a private conversation with @simonepigazzini I understood they are fine running this test in the upcoming MWGR, but otherwise they would like to keep things as they were, _i.e._ only run the pedestals with stable beams.
So I don't think including this in the cosmic scenario is the way to go. Maybe create a MWGR scenario? (if this even makes sense...)

Reply:
I don't see how your proposal differs from mine in this respect other than being worse, since it also accepts the collision in unstable beam.

Reply:
Ah I see your point now...indeed it might be a good solution to run it only for cosmics and it can still be reverted after the MWGR. I'll have a look how to customize the `cosmics.py`.

Reply:
> I understood they are fine running this test in the upcoming MWGR, but otherwise they would like to keep things as they were, i.e. only run the pedestals with stable beams.

One additional comment.
If there is concern that running the ECAL pedestal workflow on non-stable beams might produce conditions polluting the production pedestal tags, a valid option is to make this line configurable:

https://github.com/cms-sw/cmssw/blob/c861ecffe8b371654f4f0e917036d83a0801f07e/Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc#L147

so that in the case of non-stable beam configuration it writes on a different record name than `EcalPedestalsRcd` (can be whatever random string) and then use that key to steer the PCL upload to a different tag name at the level of `DropBoxMetaData` by adding an appropriate entry here (for a different record).

https://github.com/cms-sw/cmssw/blob/74144ae45f53071e611f5151557273afa4a15813/CondFormats/Common/test/ProduceDropBoxMetadata.py#L145-L149

Notice that in this context a `record` is just a dummy string and doesn't map to the actual C++ class associated to the CondFormat (as you can see in the `DropBoxMetaData` we have all sort of dummy names).

Reply:
@mmusich two further cosiderations:

1) Regarding the flag: since this is a one time only change (@simonepigazzini please confirm this from ECAL side) and will be reverted after MWGR I think it's faster to change the flag as I already did instead of modifying the cosmics customization (because I need some time to understand how to do that properly).

2) In any case, in order not to pollute the production tags, we are planning to write the payloads to a test tag as you suggest. I was under the impression that changing the `DropBoxMetaData` was enough to do that, but from your last [comment](https://github.com/cms-sw/cmssw/pull/33765#discussion_r634584347) it seems that some hardcoded changes are needed in https://github.com/cms-sw/cmssw/blob/c861ecffe8b371654f4f0e917036d83a0801f07e/Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc#L147
Could you help me understand better?

Reply:
@francescobrivio the `DropBoxMetaData` mechanism assigns the tags on which to write the payload produced by the PCL by means of reading from the framework job report the target record produced by the PCL harvester configuration. That's why the record in the source C++ code is relevant to steer which tags to populate. If you don't foresee to have a separate harvesting for cosmics then there is no need to change the source code (that would be needed to maintain at the same time two harvesting flavours), but just create a new IOV for the `DropBoxMetaData` record in the express Global Tag and then roll it back after the MWGR.
We can talk more at length offline if you wish.

Reply:
> @francescobrivio the `DropBoxMetaData` mechanism assigns the tags on which to write the payload produced by the PCL by means of reading from the framework job report the target record produced by the PCL harvester configuration. That's why the record in the source C++ code is relevant to steer which tags to populate. If you don't foresee to have a separate harvesting for cosmics then there is no need to change the source code (that would be needed to maintain at the same time two harvesting flavours), 

Thanks for the nice explanation.

> but just create a new IOV for the `DropBoxMetaData` record in the express Global Tag and then roll it back after the MWGR.

Yes, this is what I had in mind to do.

> We can talk more at length offline if you wish.

Yes, that would be really useful thanks!

So I think this PR is fine as it is now, given also that all the [tests](https://github.com/cms-sw/cmssw/pull/33765#issuecomment-843581626) are succesful.